File: Problem/Medium/101-200/198. House Robber/_c.java
// c.java
// more better version

class Solution {
	public int rob(int[] nums) {
		int first = 0;
		int second = 0;
		int temp;

		for (int val : nums) {
			temp = first;
			first = second;
			second = Math.max(second, val + temp);
		}

		return second;
	}
}

File: Problem/Medium/101-200/198. House Robber/c.py
class Solution:
    def rob(self, nums):
        n = len(nums)

        if n == 1:
            return nums[0]

        first, second = nums[0], max(nums[0], nums[1])

        for i in range(2, n):
            first, second = second, max(second, nums[i] + first)

        return second
